Content.Shared/Prototypes/LocalizedPrototype.cs

```diff
@@ -0,0 +1,35 @@
+using System.Linq;
+using Robust.Shared.Prototypes;
+
+namespace Content.Shared.Prototypes;
+
+public abstract class LocalizedPrototype : IPrototype
+{
+    [IdDataField]
+    public string ID { get; } = default!;
+
+    public const string LocFormat = "{0}-{1}-{2}";
+
+    /// <summary>The localization string for the name of this prototype</summary>
+    public string NameLoc => ToLocalizationString("name");
+    /// <summary>The localized string for the name of prototype</summary>
+    public string Name => Loc.GetString(NameLoc);
```

Comment:
This recalculates NameLoc every time either of those properties is accessed, which is bad. It should be cached and only calculated once (perhaps as a deserialization hook, or a lazily evaluated/cached property?)